libraries/Microsoft.Bot.Builder.Dialogs.Declarative/Resources/ResourceExplorer.cs

```diff
@@ -487,8 +503,12 @@ public async Task<JToken> ResolveRefAsync(JToken refToken, SourceContext sourceC
         /// <returns>The resulting <see cref="JToken"/> and <see cref="SourceRange"/> for the requested resource.</returns>
         internal async Task<(JToken, SourceRange)> ReadTokenRangeAsync(Resource resource, SourceContext sourceContext, bool advanceJsonReader = false)
         {
-            var text = await resource.ReadTextAsync().ConfigureAwait(false);
-            using (var readerText = new StringReader(text))
+            if (_cache.TryGetValue(resource.FullName, out (JToken, SourceRange) cachedResult))
```

Comment:
Still need to solve how to invalidate / update cache on resource changed event.

Reply:
Updated to refresh the cache from InterfaceConverter.